"""
Задания по исключениям.
"""
"""
1) Опишите полный синтаксис конструкции try-except:
"""
# try:
#     <body> # код с вероятным исключением 
# except:
#     <body except> сработает только если ошибка в try
# <else>:
#     <body else> сработает только если нет ошибки в try
# <finally>:
#     <body finally> сработает в любом случае
"""
2) Дан следующий код:
b = 10
c = 11
print(a)
Обработайте ошибку, которая может возникнуть.
"""
# b = 10
# c = 11
# try:
#     print(a)
# except NameError:
#     print('Name "a" not found')
"""
3) Напишите программу которая будет получать два числа через input и делить одно на другое. Обработайте ошибку, которая возникнет в случае, если второе число окажется 0 и выведите сообщение.
"""
# while True:
#     try:
#         a = int(input('a: '))
#         b = int(input('b: '))
#         print(a / b)
#         break
#     except ValueError:
#         print('Enter a num')
#     except ZeroDivisionError:
#         print('Cannot be divided by 0 (b)')
"""
4) Напишите программу, которая будет получать через инпут 2 числа и будет печатать их сумму. Обработайте ошибку, которая возникнет, если пользователь введёт не числовое значение и выведите сообщение.
"""
# while True:
#     try:
#         a = int(input('a: '))
#         b = int(input('b: '))
#         print(a / b)
#         break
#     except ValueError:
#         print('Enter a num')
#     except ZeroDivisionError:
#         print('Cannot be divided by 0 (b)')
"""
5) Дан словарь. Попытайтесь получить значение по ключу. Обработайте ошибку, возникающую в том случае, если запрашиваемый ключ не существует.
"""
# try:
#     a = {1: 'one', 2: 'two', 3: 'three'}
#     print(a[4])
# except KeyError:
#     print('Key not found')
"""
6)  Дан список. Обработайте исключение при попытке обращения к несуществующему элементу.
"""
# try:
#     a = [1, 2, 3]
#     print(a[3])
# except IndexError:
#     print('Index not found')
"""
7) Попытайтесь в блоке try принять 2 числа и разделить одно на другое. В блоке except обработайте сразу 2 возможных исключения.
"""
# while True:
#     try:
#         a = int(input('a: '))
#         b = int(input('b: '))
#         print(a / b)
#         break
#     except ValueError:
#         print('Enter a num')
#     except ZeroDivisionError:
#         print('Cannot be divided by 0 (b)')
"""
8) Запросите у пользователя сумму, которая у него сейчас есть в бумажнике. Если он введёт сумму, меньшую чем 0, то выбросите исключение с текстом "Сумма не может быть отрицательной!"
"""
# sum = int(input())
# try:
#     if sum < 0:
#         raise
# except RuntimeError:
#     print("Сумма не может быть отрицательной!")
"""
9)  В блоке try запросите у пользователя ввод его возраста. Затем в том же блоке проверьте его возраст на совершеннолетие. Если пользователь несовершеннолетний, выбросите исключение с текстом "Доступ запрещён". Обработайте также исключение если пользователь вводит текст вместо числа с сообщением 'Введен некорректный возраст'. Если ошибок не возникло, то распечатайте сообщение "Спасибо!", а затем "До свидания", вне зависимости от того, произошла ошибка или нет.
"""
# try:
#     a = int(input('Enter your age: '))
#     if a < 18:
#         raise
# except RuntimeError:
#     print('Доступ запрещен')
# except ValueError:
#     print('Введен некорректный возраст')
# else:
#     print('Спасибо!')
# finally:
#     print('До свидания')
